File: typescript/src/adapters/openai.ts
import { Message, Context } from '../core/models.js';

export interface OpenAIMessage {
  role: string;
  content: string;
  name?: string;
  tool_calls?: Array<Record<string, unknown>>;
  tool_call_id?: string;
}

export class OpenAIAdapter {
  static toMessages(context: Context): OpenAIMessage[] {
    return context.messages.map((msg: Message) => {
      const result: OpenAIMessage = {
        role: msg.role,
        content: msg.content,
      };

      if (msg.name !== undefined) {
        result.name = msg.name;
      }

      if (msg.toolCalls !== undefined) {
        result.tool_calls = msg.toolCalls;
      }

      if (msg.toolCallId !== undefined) {
        result.tool_call_id = msg.toolCallId;
      }

      return result;
    });
  }
}

File: python/src/agent_context/adapters/openai.py
from __future__ import annotations

from agent_context.core.models import Context


class OpenAIAdapter:
    """Adapter for converting Context to OpenAI-compatible message format."""

    @staticmethod
    def to_messages(context: Context) -> list[dict]:
        """Convert a Context to a list of OpenAI-compatible message dicts."""
        result = []
        for msg in context.messages:
            d = {
                "role": msg.role,
                "content": msg.content,
            }
            if msg.name is not None:
                d["name"] = msg.name
            if msg.tool_calls is not None:
                d["tool_calls"] = msg.tool_calls
            if msg.tool_call_id is not None:
                d["tool_call_id"] = msg.tool_call_id
            result.append(d)
        return result
